ckanext/xloader/plugin.py

```diff
@@ -183,3 +199,39 @@ def get_helpers(self):
             "xloader_status_description": xloader_helpers.xloader_status_description,
             "is_resource_supported_by_xloader": xloader_helpers.is_resource_supported_by_xloader,
         }
+
+
+def _should_remove_unsupported_resource_from_datastore(res_dict):
```

Comment:
This is very complex logic.

Can you add some low level unit tests to cover most of the logic paths, the return statement has around 32 logic flows alone. 

maybe add to 
https://github.com/qld-gov-au/ckanext-xloader/blob/master/ckanext/xloader/tests/test_plugin.py
or another test file next to it.


I really want to ensure that this is not be too greedy and delete 'url' resources which can be loaded into the datastore.